### PR TITLE
nixos/security.sudo-rs: remove `with lib;`

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #208242

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/security/sudo-rs.nix [hunk 1]
```diff
@@ -76,15 +73,15 @@ in
       '';
     };
 
-    extraRules = mkOption {
+    extraRules = lib.mkOption {
       description = ''
         Define specific rules to be in the {file}`sudoers` file.
         More specific rules should come after more general ones in order to
-        yield the expected behavior. You can use mkBefore/mkAfter to ensure
+        yield the expected behavior. You can use lib.mkBefore/mkAfter to ensure
```
Comment (philiptaron) [inline nixos/modules/security/sudo-rs.nix:80]: revert

File: nixos/modules/security/sudo-rs.nix [hunk 2]
```diff
@@ -200,35 +197,35 @@ in
             options = opts ++ cfg.defaultOptions;
           } ];
         } ];
-      in mkMerge [
-        # This is ordered before users' `mkBefore` rules,
+      in lib.mkMerge [
+        # This is ordered before users' `lib.mkBefore` rules,
```
Comment (philiptaron) [inline nixos/modules/security/sudo-rs.nix:201]: questionable, lean to keep

File: nixos/modules/security/sudo-rs.nix [hunk 3]
```diff
@@ -200,35 +197,35 @@ in
...
         # so as not to introduce unexpected changes.
-        (mkOrder 400 (defaultRule { users = [ "root" ]; }))
+        (lib.mkOrder 400 (defaultRule { users = [ "root" ]; }))
 
         # This is ordered to show before (most) other rules, but
-        # late-enough for a user to `mkBefore` it.
-        (mkOrder 600 (defaultRule {
+        # late-enough for a user to `lib.mkBefore` it.
```
Comment (philiptaron) [inline nixos/modules/security/sudo-rs.nix:206]: questionable, lean to keep

**Review by philiptaron — CHANGES_REQUESTED** [review]

File: nixos/modules/security/sudo-rs.nix (hunk 1 above)
Comment (Stunkymonkey) [reply nixos/modules/security/sudo-rs.nix:80]: really? maybe even add `lib.mkAfter`?

Comment (philiptaron) [reply nixos/modules/security/sudo-rs.nix:80]: Put it in backticks (`lib.mkBefore`/`lib.mkAfter`) and you have yourself a deal.

**Review by philiptaron — APPROVED** [review]

**Pull request merged**